File: AFMobile/AFWinPhone/AFWinPhone/parsers/JSONParser.cs
using System;
using AFWinPhone.components.parts;
using Windows.Data.Json;

namespace AFWinPhone.parsers
{
    /// <summary>
    /// Interface for JSON parsers of components information.
    /// </summary>
    public interface JSONParser
    {
        /// <summary>
        /// arses given string in JSON format. Parsing should be different if parsing inner class of component.
        /// </summary>
        /// <param name="jsonStrToBeParsed">string to be parsed</param>
        /// <param name="parsingInnerClass">true if parsing inner class of component</param>
        /// <returns>information wrapped in object</returns>
        AFClassInfo parse(String jsonStrToBeParsed, bool parsingInnerClass);
    }
}

File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/parsers/JSONParser.java
package cz.cvut.fel.matyapav.afandroid.parsers;

/**
 * Interface for JSON parsers of components information.
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz)
 *
 * @since 1.0.0.
 */
public interface JSONParser<T> {

    /**
     * Parses given string in JSON format. Parsing should be different if parsing inner class of component.
     *
     * @param jsonStrToBeParsed string to be parsed
     * @param parsingInnerClass true if parsing inner class of component
     * @return information wrapped in some object
     */
    T parse(String jsonStrToBeParsed, boolean parsingInnerClass);
}
